### PR TITLE
2326 drc debugs set team teamtype endpoint

### LINKED ISSUE #2326: DRC Debugs - Set Team, TeamType Endpoint
### Description

We want to create an endpoint that allows a user to set the team type on a team ('electrical, business, software, or mechanical'). This ticket is to do the schema migration to allow that and to make the associated endpoint

### Acceptance Criteria

- [x] You should edit the schema so teams have a teamType attribute (should be optional)
- [x] This endpoint should be on the route 'teams/:teamId/set-team-type'
- [x] The payload should be a teamType
- [x] Only admins should be able to set the team type on a team
- [x] Manually test using Postman, and include screenshots of query and result (reach out to Sean if you have questions)

### PR DESCRIPTION
**Pull request by AShyamkumar19**
## Changes

Created a setTeamType endpoint, where the admin can set a team type for each team in its project. 

## Screenshots

This adds the teamType to a Team:
image

The added teamType in prisma studio:
Screenshot 2024-04-13 162903

Testing for exceptions:

Wrong teamTypeId
image

Wrong Team
image

Wrong user (not admin)
image

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Screenshots of UI changes (see Screenshots section)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes #2326

### CODE REVIEW

File: src/backend/src/controllers/teams.controllers.ts [hunk 1]
```diff
@@ -127,4 +127,17 @@ export default class TeamsController {
       next(error);
     }
   }
+
+  static async setTeamType(req: Request, res: Response, next: NextFunction) {
+    try {
+      const { teamTypeId } = req.body;
+      const submitter = await getCurrentUser(res);
+
+      const updateTeamType = await TeamsService.setTeamType(submitter, req.params.teamId, teamTypeId);
```
Comment (Aaryan1203) [inline src/backend/src/controllers/teams.controllers.ts:136]: can you destructure the teamId just like you do it for the teamTypeId, it just makes it a bit easier to read

File: src/backend/src/services/teams.services.ts [hunk 2]
```diff
@@ -372,4 +372,38 @@ export default class TeamsService {
...
+      ...teamQueryArgs
+    });
+
+    if (!team) {
+      throw new NotFoundException('Team', teamId);
+    }
+
+    if (!teamType) {
+      throw new HttpException(400, 'Team type with the given ID does not exist');
+    }
```
Comment (Aaryan1203) [inline src/backend/src/services/teams.services.ts:396]: can you move this check right after you define the teamType constant

File: src/backend/src/services/teams.services.ts [hunk 3]
```diff
@@ -372,4 +372,38 @@ export default class TeamsService {
...
+      where: { teamId },
+      ...teamQueryArgs
+    });
+
+    if (!team) {
+      throw new NotFoundException('Team', teamId);
+    }
+
+    if (!teamType) {
+      throw new HttpException(400, 'Team type with the given ID does not exist');
```
Comment (Aaryan1203) [inline src/backend/src/services/teams.services.ts:395]: can you throw a not found exception here as well

**Review by Aaryan1203 — CHANGES_REQUESTED** [review]
Great work on your ticket! Just left a few comments

Can you also add the postman screenshots of the request you send to set the teamtype and the response you get back from it. in the screenshot I don't think I saw the set team type on the team

**Review by Aaryan1203 — CHANGES_REQUESTED** [review]
Also can you make your pr more descriptive, a lot of the parts of the template are left as original

File: src/backend/src/controllers/teams.controllers.ts [hunk 4]
```diff
@@ -127,4 +127,18 @@ export default class TeamsController {
...
     }
   }
+
+  static async setTeamType(req: Request, res: Response, next: NextFunction) {
+    try {
+      const { teamTypeId } = req.body;
+      const { teamId } = req.params;
+      const submitter = await getCurrentUser(res);
+
+      const updateTeamType = await TeamsService.setTeamType(submitter, teamId, teamTypeId);
```
Comment (RChandler234) [inline src/backend/src/controllers/teams.controllers.ts:137]: updatedTeam*

File: src/backend/src/services/teams.services.ts [hunk 5]
```diff
@@ -372,4 +372,38 @@ export default class TeamsService {
     const teamTypes = await prisma.teamType.findMany();
     return teamTypes;
   }
+
+  static async setTeamType(submitter: User, teamId: string, teamTypeId: string): Promise<Team> {
```
Comment (RChandler234) [inline src/backend/src/services/teams.services.ts:376]: you should add a thorough comment to this

File: src/backend/src/services/teams.services.ts [hunk 6]
```diff
@@ -372,4 +372,38 @@ export default class TeamsService {
...
+    const team = await prisma.team.findUnique({
+      where: { teamId },
+      ...teamQueryArgs
+    });
+
+    if (!team) {
+      throw new NotFoundException('Team', teamId);
+    }
+
+    const updateTeam = await prisma.team.update({
```
Comment (RChandler234) [inline src/backend/src/services/teams.services.ts:398]: updated*

**Review by RChandler234 — CHANGES_REQUESTED** [review]
looking really good!

**Pull request merged**